### PR TITLE
Shreyas/warehouse history clean

### PR DESCRIPTION
**Pull request by shreyas-thirumale**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

### Code Review Preparation
- [X] I have performed a self-review and reviewed the changed files
- [X] My code follows the project's style guidelines and passes linting
- [X] I have added comments in hard-to-understand areas and updated documentation

### Testing (Ignore for the first PR)
- [X] I have added relevant tests and they pass locally
- [X] I have tested the changes manually, including edge cases

### Code Quality & Security
- [X] I have formatted my code and removed debugging/unused code
- [X] No TypeScript errors or linting issues
- [X] No sensitive information committed (API keys, passwords, etc.)
- [X] User inputs are validated and sanitized

### Dependencies & Deployment
- [X] Dependencies are documented and package-lock.json is updated if needed
- [X] Changes work with Firebase emulator and environment variables are documentedRetry

## Screenshots (If it is a front end feature screenshot is required)

image

## Release Notes

* **New Features**
  * Control Panel interface with navigation tabs for Overview and Warehouse History sections
  * Warehouse History page displaying inventory change records with multi-criteria filtering
  * Search functionality across inventory items, users, and change types
  * Sort options by date and time for viewing change history
  * Revert capability for inventory changes with confirmation dialogs (note: Delete operations cannot be reverted)

### CODE REVIEW

File: components/control-panel/WarehouseHistoryTable.tsx
Comment (joelchem) [inline components/control-panel/WarehouseHistoryTable.tsx]: this component should be refactored to match the current version in the figma design

File: app/control-panel/warehouse-history/page.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            });
+        }
+
+        return result;
+    }, [changes, appliedSearch, selectedTypes, dateSortStatus, timeSortStatus]);
+
+    return (
+        <>
+            <div className="flex flex-wrap gap-3 mb-4 shrink-0">
+                <SearchBox
```
Comment (joelchem) [inline app/control-panel/warehouse-history/page.tsx]: for the searching patterns you can just copy how its done on other pages (like donation review)

**Review by joelchem — CHANGES_REQUESTED** [review]
good work! generally some architectural changes. i think as we mentioned there are a lot of remnants from old commits persisting. try to see what you can do to get rid of these remnants because they're most likely just going to make things more confusing.

this kinda chopped but one possible way would be to just make a new branch off the latest main and manually copy paste over the components/pages you edited and then making it work in the new branch.

we're going to get out updated types for InventoryChange that should be the exact one you should use. It's going to include a "reverted" column that indicates whether a certain InventoryChange has been reverted or not. These types, a long with test data will be added by @constrictingsnake and merged into main soon.

if you need any clarifications with any of that be sure to let us know!

**Comment by joelchem** [comment]
Moved to #87
